src/oscar/apps/basket/forms.py

```diff
@@ -182,8 +219,8 @@ def _add_option_field(self, product, option):
         This is designed to be overridden so that specific widgets can be used
         for certain types of options.
         """
-        self.fields[option.code] = forms.CharField(
-            label=option.name, required=option.is_required)
+        option_field = self.OPTION_FIELD_FACTORIES[option.type](option)
```

Comment:
I think we need to default to the charfield if the option type is not found in the map - this can happen if a project overrides the list of types.